### PR TITLE
Makeing it work with Gnome 40. #340

### PR DESCRIPTION
**Pull request by jnphilipp**

### CODE REVIEW

**Review by hedayat — APPROVED** [review]
Thank you for the contribution.

The code looks fine to me, and it also works fine. 

~LGTM~
Please mark it as backward compatible if it is so. Thanks!

File: data/metadata.json.in [hunk 1]
```diff
@@ -10,8 +10,7 @@
     "gettext-domain": "hamster-shell-extension",
     "settings-schema": "org.gnome.shell.extensions.project-hamster",
     "shell-version": [
-        "3.34",
-        "3.36"
+        "40"
```
Comment (hedayat) [inline data/metadata.json.in:16]: But, have you tested the extension on older gnome shell versions? I guess it should also work on older gnome shell versions. If so, please retain their compatibility string here, even though it is ignored by some of them.

Comment (hedayat) [reply data/metadata.json.in:16]: Also I've merged another small change which added 3.38 to the list. Please fix the conflict too. Thanks!

**Comment by hedayat** [comment]
Dear @jnphilipp ,
would you please resolve the conflict and let me know if this extension is compatible with older versions?

**Comment by jnphilipp** [comment]
Needed to make some changes, to get it to work on my Ubuntu 20.04 with 3.36. Haven't tested it on other versions.

**Comment by hedayat** [comment]
> Needed to make some changes, to get it to work on my Ubuntu 20.04 with 3.36. Haven't tested it on other versions.

Thanks a lot. I think it is sufficient :)

**Comment by mwilck** [comment]
@jnphilipp, @hedayat, thanks for your work on this, highly appreciated.

Allow me to make a critical remark nonetheless. I wouldn't have acked this if I had looked at it before it was merged. Being late to the party, I can't seriously complain, but please mind the following for future PRs:

 * There are non-trivial code changes in the merge commit 89a93f4, in `extension/prefs.js`. Apparently you added `visible: true` properties to various widgets. That was probably a correct change, but changes like this should go into a separate commit with appropriate message, rather than being hidden in a merge commit. Moreover, in a PR branch you should rebase the PR if upstream changes, rather than merging. It makes reading the history much easier later on.
 * The header of the commit message is wrong English. This is just a nitpick, but it hurts my eye when  I look at the commit history. I guess both of you aren't native English speakers. Me neither. Yet everyone can avoid mistakes like this simply by using  a spell checker.

**Comment by hedayat** [comment]
OK. 

To be honest, I didn't care about the merge instead of rebase thing, and while I myself usually do that, I'm not sure if it is good thing to **require** it. 

Having separate commits is OK, I am just not knowledgeable enough about this JS/shell thing to decide about the boundries. :P  

About the header, well, I'm really sorry, I'm not sure if I've even read it carefully back then :D **Makeing** is just too weird to even need a spell checker :P

**Pull request merged**